test/mocks/v1/non_template_tests_test.go

```diff
@@ -0,0 +1,241 @@
+			Crd:         AnotherMockResourceCrd,
+			Cfg:         cfg,
+			SharedCache: kuberc.NewKubeCache(context.TODO()),
+		}
+
+		err = helpers.AddAndRegisterCrd(ctx, AnotherMockResourceCrd, clientset)
+		Expect(err).NotTo(HaveOccurred())
+
+		anotherMockResourceClient, err = NewAnotherMockResourceClient(ctx, anotherMockResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// ClusterResource Constructor
+		clusterResourceClientFactory := &factory.KubeResourceClientFactory{
+			Crd:         ClusterResourceCrd,
+			Cfg:         cfg,
+			SharedCache: kuberc.NewKubeCache(context.TODO()),
+		}
+
+		err = helpers.AddAndRegisterCrd(ctx, ClusterResourceCrd, clientset)
+		Expect(err).NotTo(HaveOccurred())
+
+		clusterResourceClient, err = NewClusterResourceClient(ctx, clusterResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// MockCustomType Constructor
+		mockCustomTypeClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		mockCustomTypeClient, err = NewMockCustomTypeClient(ctx, mockCustomTypeClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// Pod Constructor
+		podClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		podClient, err = github_com_solo_io_solo_kit_pkg_api_v1_resources_common_kubernetes.NewPodClient(ctx, podClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		emitter = NewTestingEmitter(simpleMockResourceClient, mockResourceClient, fakeResourceClient, anotherMockResourceClient, clusterResourceClient, mockCustomTypeClient, podClient)
+
+		// create `FakeResource`s in "namespace1" and "slowWatchNamespace"
+		fakeResource1a, err = fakeResourceClient.Write(NewFakeResource(namespace1, name1), clients.WriteOpts{Ctx: ctx})
+		Expect(err).NotTo(HaveOccurred())
+		fakeResource1b, err = fakeResourceClient.Write(NewFakeResource(slowWatchNamespace, name1), clients.WriteOpts{Ctx: ctx})
+		Expect(err).NotTo(HaveOccurred())
+	})
+	AfterEach(func() {
+		kubeutils.DeleteNamespacesInParallelBlocking(ctx, kube, namespace1, slowWatchNamespace)
+
+		// clean up created resources
+		err := fakeResourceClient.Delete(fakeResource1a.GetMetadata().Namespace, fakeResource1a.GetMetadata().Name, clients.DeleteOpts{Ctx: ctx})
+		Expect(err).NotTo(HaveOccurred())
+		err = fakeResourceClient.Delete(fakeResource1b.GetMetadata().Namespace, fakeResource1b.GetMetadata().Name, clients.DeleteOpts{Ctx: ctx})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Should not overwrite initial listed resources for non-returned namespace watchers", func() {
+		/*
+			This test relies on some hardcoded logic in `resource_client_template.go`.  In particular:
+			`FakeResourceClient`, when used in conjunction with the `slowWatchNamespace` has an artificial
+			5s delay when getting back the results from .Watch.  The idea behind the test is that
+			we have a normal, unimpeded watch running in parallel with a slow watch to simulate a race condition.
```

Comment:
I think we can remove this comment, and probably move some of it to the implementation of our slowFakeResourceClient, to include context around what it is, why it exists